File: Dataset/Leetcode/train/69/439.cpp
class Solution {
public:
    int XXX(int y) {
        if (y==0 || y==1) return y;
        long int x1 = y/2;
        long int x;
        while (true) {
            x = (y + x1 * x1) / (2 * x1);
            if ((x*x) > y && (x-1)*(x-1) <= y) return x-1;
            if ((x*x) <= y && (x+1)*(x+1) > y) return x;
            x1 = x;
        }
        return -1;
    }
};

File: Dataset/Leetcode/train/121/439.java
 class Solution {
    public int XXX(int[] prices) {
        int maxpro = 0;
        for(int i = 0; i < prices.length-1; i++){
            for(int j = i +1; j < prices.length;j++){
                if(prices[i]-prices[j] < 0 && maxpro-(prices[j] - prices[i]) < 0){
                    maxpro = prices[j]-prices[i];
                }
                
            }
        }
        return maxpro;
    }
}

